### PR TITLE
Move _DiskArray out of tests

### PR DESCRIPTION
**Pull request by meggart**

### CODE REVIEW

File: Manifest.toml
Comment (felixcremer) [inline Manifest.toml:1]: Why are you commiting the Manifest.toml?

File: src/util/testtypes.jl [hunk 1]
```diff
@@ -0,0 +1,59 @@
+module TestTypes
+export _DiskArray, UnchunkedDiskArray, getindex_count, setindex_count, trueparent
+import ..DiskArrays
+
+# Define a data structure that can be used for testing
+struct _DiskArray{T,N,A<:AbstractArray{T,N}} <: AbstractArray{T,N}
```
Comment (rafaqz) [inline src/util/testtypes.jl:6]: Call this `AccessCountDiskArray` ?

File: Manifest.toml
Comment (meggart) [reply Manifest.toml:1]: Accident

File: src/util/testtypes.jl [hunk 2]
```diff
@@ -0,0 +1,80 @@
+module TestTypes
+
+import ..DiskArrays
+
+export AccessCountDiskArray, UnchunkedDiskArray, getindex_count, setindex_count, trueparent
+
+"""
+    AccessCountDiskArray(A; chunksize)
+    
+An array that counte `getindex` and `setindex` calls, to debug
```
Comment (felixcremer) [inline src/util/testtypes.jl:10]:
```suggestion
An array that count `getindex` and `setindex` calls, to debug
```

**Pull request merged**